pkg/utils.go

```diff
@@ -0,0 +1,64 @@
+package pkg
+
+import ("time"
+		"encoding/json"
+	)
+
+
+
+//function which clean history within a given window in Hours
+//if we buy coffe each 23 hours user history  won't be evicted from
+//redis but we would have stored data about orders which are not in last 24 hours
+func CleanWithinWindow(history *OrderHistory, window int) OrderHistory {
+	end := time.Now()
+	start := end.Add(-(time.Duration(window) * time.Hour))
+
+	filtered := []Order{}
+
+	for _,order := range history.Orders {
+        if InTimeSpan(start,end,order.Timestamp){
+			filtered  = append(filtered,order)
+		}
+    }
```

Comment:
(very minor) please, run `go fmt` command before every commit. It will format your go code

Reply:
got it